pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -222,6 +240,44 @@ func (f *framework) RunPrefilterPlugins(
 	return nil
 }
 
+// RunScorePlugins runs the set of configured scoring plugins. It returns a map that
+// stores for each scoring plugin name the corresponding NodeScoreList(s).
+// It also returns *Status, which is set to non-success if any of the plugins returns
+// a non-success status.
+func (f *framework) RunScorePlugins(pc *PluginContext, pod *v1.Pod, nodes []*v1.Node) (PluginToNodeScoreMap, *Status) {
+	pluginToNodeScoreMap := make(PluginToNodeScoreMap, len(f.scorePlugins))
+	for _, pl := range f.scorePlugins {
+		var (
+			mu      = sync.Mutex{}
+			errMsgs []string
```

Comment:
It seems like we could concern the first error, we could make an err channel like this:

```go
errCh := make(chan string, len(nodes))
```

Reply:
merged into https://github.com/kubernetes/kubernetes/pull/79109#discussion_r298857399